src/roles/check_database_connection/tasks/main.yaml

```
@@ -0,0 +1,17 @@
+---
+- name: Check Foreman database connectivity and credentials
+  community.postgresql.postgresql_ping:
+    login_host: "{{ foreman_database_host }}"
+    login_user: "{{ foreman_database_user }}"
+    login_password: "{{ foreman_database_password }}"
+    login_db: "{{ foreman_database_name }}"
+  register: check_database_connection_ping_result
+  ignore_errors: true
+
+- name: Assert Foreman database is reachable
+  ansible.builtin.assert:
+    that:
+      - check_database_connection_ping_result.ping | default(false)
```

Shouldn't that be 
```suggestion
      - check_database_connection_ping_result.is_available
```
at least that's how I read the code of the module